azure/services/scalesets/scalesets.go

```diff
@@ -673,27 +578,57 @@ func (s *Service) getVirtualMachineScaleSetDefaultNetworkConfiguration(vmssSpec
 				})
 		}
 	}
-	return &[]compute.VirtualMachineScaleSetNetworkConfiguration{{
-		Name: pointer.String(vmssSpec.Name),
-		VirtualMachineScaleSetNetworkConfigurationProperties: &compute.VirtualMachineScaleSetNetworkConfigurationProperties{
-			Primary:            pointer.Bool(true),
-			EnableIPForwarding: pointer.Bool(true),
-			IPConfigurations: &[]compute.VirtualMachineScaleSetIPConfiguration{
-				{
-					Name: pointer.String(vmssSpec.Name),
-					VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
-						Subnet: &compute.APIEntityReference{
-							ID: pointer.String(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, vmssSpec.SubnetName)),
-						},
-						Primary:                         pointer.Bool(true),
-						PrivateIPAddressVersion:         compute.IPVersionIPv4,
-						LoadBalancerBackendAddressPools: &backendAddressPools,
+	nicConfigs := []compute.VirtualMachineScaleSetNetworkConfiguration{}
+	for i, n := range vmssSpec.NetworkInterfaces {
+		nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
+		nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
+		nicConfig.Name = pointer.String(vmssSpec.Name + "-nic-" + strconv.Itoa(i))
+		if pointer.BoolDeref(n.AcceleratedNetworking, false) {
+			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(true)
+		} else {
+			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(false)
+		}
+
+		// Create IPConfigs
+		ipconfigs := []compute.VirtualMachineScaleSetIPConfiguration{}
+		for j := 0; j < n.PrivateIPConfigs; j++ {
+			ipconfig := compute.VirtualMachineScaleSetIPConfiguration{
+				Name: pointer.String(fmt.Sprintf("ipConfig" + strconv.Itoa(j))),
```

Comment:
I'm not sure about the semantics of the IPConfiguration API resource. Is it namespaced below the nic config and the name does therefore not matter or do we need to include e.g. `vmssSpec.Name` in the name to avoid name collisions? 

Asking because previous code used the vmss name in the IP configuration too.